### PR TITLE
feat: persist compaction time window

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR: 
- persist time window durign compaction and write it to region manifest so that it can be recovered during manifest replay
- move compaction window option from `RegionMetdata` to `LevelMetas` struct so that in the future it can accommodate per-level compaction window, which is required by twcs
- remove compaction time window table/region option, as we never actually announce this option so it does not break compatibility.

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/storage/src/compaction/task.rs [hunk 1]
```diff
@@ -151,7 +151,10 @@ impl<S: LogStore> CompactionTask for CompactionTaskImpl<S> {
 
         let input_ids = compacted.iter().map(|f| f.file_id).collect::<Vec<_>>();
         let output_ids = output.iter().map(|f| f.file_id).collect::<Vec<_>>();
-        info!("Compacting SST files, input: {input_ids:?}, output: {output_ids:?}");
+        info!(
+            "Compacting SST files, input: {input_ids:?}, output: {output_ids:?}, window: {:?}",
+            self.compaction_time_window
```
Comment (evenyag) [inline src/storage/src/compaction/task.rs:156]: nitpick: Consider using the same style
```suggestion
            "Compacting SST files, input: {:?}, output: {:?}, window: {:?}",
            input_ids, output_ids, self.compaction_time_window
```

File: src/storage/src/manifest/action.rs [hunk 2]
```diff
@@ -38,8 +38,6 @@ pub struct RawRegionMetadata {
     pub columns: RawColumnsMetadata,
     pub column_families: RawColumnFamiliesMetadata,
     pub version: VersionNumber,
-    /// Time window for compaction
```
Comment (evenyag) [inline src/storage/src/manifest/action.rs:41]: Document the precision of this timestamp.

File: src/storage/src/region/tests/compact.rs [hunk 3]
```diff
@@ -220,6 +226,42 @@ impl CompactionTester {
...
     }
+
+    async fn reopen(&mut self) -> Result<bool> {
+        // Close the old region.
+        if let Some(base) = self.base.take() {
+            base.close().await;
+        }
+
+        // Reopen the region.
+        let object_store = new_object_store(&self.store_dir, None);
```
Comment (evenyag) [inline src/storage/src/region/tests/compact.rs:237]: Is it possible to reuse other helper function to construct the region?

File: src/storage/src/sst.rs [hunk 4]
```diff
@@ -65,6 +65,7 @@ pub struct LevelMetas {
     levels: LevelMetaVec,
     sst_layer: AccessLayerRef,
     file_purger: FilePurgerRef,
+    compaction_window: Arc<RwLock<Option<i64>>>,
```
Comment (evenyag) [inline src/storage/src/sst.rs:68]: I think we should use a copy-on-write pattern to update the compaction window like how we update the `LevelMetas`.

File: src/storage/src/sst.rs [hunk 5]
```diff
@@ -91,6 +93,15 @@ impl LevelMetas {
         self.levels.len()
     }
 
+    pub fn set_compaction_time_window(&self, window: i64) {
```
Comment (evenyag) [inline src/storage/src/sst.rs:96]: Maybe we could update the window in `merge()`. Like how we update `flushed_sequence` in `apply_edit`
```rust
pub fn merge(
    &self,
    files_to_add: impl Iterator<Item = FileMeta>,
    files_to_remove: impl Iterator<Item = FileMeta>,
    compaction_window: Option<i64>,
) -> LevelMetas {}
```

File: src/storage/src/manifest/action.rs (hunk 2 above)
Comment (v0y4g3r) [reply src/storage/src/manifest/action.rs:41]: > Document the precision of this timestamp.

This line was deleted

File: src/storage/src/region/tests/compact.rs (hunk 3 above)
Comment (v0y4g3r) [reply src/storage/src/region/tests/compact.rs:237]: Looks like it's not used by other cases, so I did not extract it to a new function.

**Comment by v0y4g3r** [comment]
@evenyag PTAL

**Review by evenyag — APPROVED** [review]
LGTM

**Review by fengjiachun — APPROVED** [review]
lgtm

**Pull request merged**